### PR TITLE
Feature/upgrade to babel7

### PR DESCRIPTION
**Pull request by unprofessional**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
- Upgraded all Babel packages to Babel7
- Upgraded `react-scripts` to latest version (see Background/Context below)

#### Related JIRA tickets:
- MERRY CHRISTMAS (did this just because the old stuff offended me too much -- feel free to make a ticket and assign it to me if you wanna track this)

#### How should this be manually tested?
- Test. Everything.
  - Literally everything should be regression tested. There was a lot to update here and it all needs to be rigorously tested to ensure the integrations didn't break some obscure edge-case.

#### Background/Context
- With respect to `react-scripts` -- this one is important:
  - `react-scripts` is basically the "ez launch" way to do things from React Create. It actually pulls in and manages dependent packages on its own -- ones you don't need to include in your own `package.json`. This is why I deleted some packages that are necessary like `webpack` where it may seem that I've broken the code -- `react-scripts` pulls in its own sub-modules with the versions it likes.
  - Eventually, if you want more control over your package management, you can do `react-scripts eject` which it seems Chris thought ahead of and added the `yarn eject` script. Note that you'd better be sure you actually wanna do it, since it's irreversible. But it basically adds all of its managed dependencies to your `package.json` including all of the script-config references (i.e. `jest` etc).
- I fixed some build warnings about `export 'default' (reexported as '**Actions') was not found in './actions/` since a lot of your `actions.js` files were exporting `const`s and `function`s but no `default` (which is where you'd usually export a big giant object instead of what you've got going on now). Just did this so it would stop complaining about them, but be aware that now this means you're *actually* exporting all your actions externally. I'm leaving it up to you on how you wanna handle that, or leave it be, etc.

### CODE REVIEW

**Comment by unprofessional** [comment]
Probably the biggest thing I can recommend for build and load times is to leverage webpack's code splitting and splitChunks properties to prevent dependency duplication.

https://webpack.js.org/guides/code-splitting/

It basically comes down to splitting up your imports, code, etc into separate bundles (where I'd organize by priority) and telling webpack to load them separately and prune out any duplicates it finds.

File: package.json [hunk 1]
```diff
@@ -1,89 +1,98 @@
...
-    "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject",
-    "eslint": "./node_modules/.bin/eslint --fix src",
-    "stylelint": "./node_modules/.bin/stylelint \"src/styles/**/*.scss\""
-  }
+    "name":"client-boilerplate",
+    "version":"1.1.2",
+    "private":true,
+    "engines":{
+        "node":">=10.11.0",
```
Comment (jmfadeley) [inline package.json]: Is there no chance we could keep this at node 8.10? That kind of version jumping risks problems.

Comment (jmfadeley) [reply package.json]: Mm, disregard. Apparently there are necessities for at least 10 in the build. But at the very least, can we force use of Node 10.15 as that's stable?

**Comment by jmfadeley** [comment]
Password reset failure is a known issue related to INBA-948. Which I need to check in.

File: package.json [hunk 2]
```diff
@@ -1,89 +1,98 @@
...
-  "scripts": {
-    "start": "node index.js",
-    "develop": "webpack-dev-server --config webpack.dev.js",
-    "build": "webpack --config webpack.prod.js",
-    "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject",
-    "eslint": "./node_modules/.bin/eslint --fix src",
-    "stylelint": "./node_modules/.bin/stylelint \"src/styles/**/*.scss\""
-  }
+    "name": "client-boilerplate",
```
[the remark this answers is not in the document]
Comment (jmfadeley) [reply package.json]: A rename is a good idea. I would prefer `indaba-client` as to match the repo whenever reasonable, as we do in most of our other (current) projects.
